### PR TITLE
Added example bug to hello world system

### PR DESCRIPTION
**Pull request by afsafzal**
`rosrun rospy_tutorials listener` & `rosbag play example.bag` will result in `listener` node throwing exceptions periodically.

### CODE REVIEW

**Comment by afsafzal** [comment]
This is a very stupid bug and it doesn't result in a node crash for the buggy node.

**Comment by ChrisTimperley** [comment]
A bug that crashes the node would be best.

File: test/hello_world/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
- && apt-get install -y \
+ && apt-get install -y build-essential qtbase5-dev \
       ros-indigo-ros-tutorials \
       ros-indigo-common-tutorials \
       vim \
       wget \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists
+
+RUN mkdir /root/overlay_ws \
```
Comment (ChrisTimperley) [inline test/hello_world/Dockerfile]: Can you rename `/root/overlay_fs` to `/ros_ws`? Also, `ROS_WSPACE` could be lifted to before this `RUN` command.

File: test/hello_world/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+    && cd /root/overlay_ws \
+    && wstool init src \
+    && cd /root/overlay_ws/src \
+    && wstool set ros_tutorials --git git://github.com/ros/ros_tutorials.git -y \
+    && wstool update 
+
+RUN cd /root/overlay_ws \
```
Comment (ChrisTimperley) [inline test/hello_world/Dockerfile]: Also, this could be folded into the `RUN` command above.

**Comment by ChrisTimperley** [comment]
> A bug that crashes the node would be best.

And a C++ example would be even better.

File: test/hello_world/Dockerfile [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
 WORKDIR "${ROS_WSPACE}"
 RUN echo "#!/bin/bash \n\
 set -e \n\
-source \"/opt/ros/indigo/setup.bash\" \n\
+source \"/opt/ros/${ROS_DISTRO}/setup.bash\" \n\
 source \"${ROS_WSPACE}/devel/setup.bash\" \n\
 exec \"\$@\"" > "${ROS_WSPACE}/entrypoint.sh" \
  && chmod +x "${ROS_WSPACE}/entrypoint.sh"
-ENTRYPOINT ["/root/overlay_ws/entrypoint.sh"]
+ENTRYPOINT ["/ros_ws/entrypoint.sh"]
```
Comment (ChrisTimperley) [inline test/hello_world/Dockerfile]: Should this be `${ROS_SPACE}`?

Comment (afsafzal) [reply test/hello_world/Dockerfile]: If I set it to `ENTRYPOINT ["${ROS_WSPACE}/entrypoint.sh"]` it gives me error that it doesn't recognize `${ROS_WSPACE}`

Comment (ChrisTimperley) [reply test/hello_world/Dockerfile]: Interesting. Okay. `/ros_ws/` is fine for now.

File: test/hello_world/Dockerfile [hunk 4]
```diff
@@ -1,9 +1,34 @@
-FROM ros:indigo-ros-base
+FROM ros:indigo
```
Comment (ChrisTimperley) [inline test/hello_world/Dockerfile]: Both of these Dockerfiles can be collapsed into one by using a multi-stage build: https://docs.docker.com/develop/develop-images/multistage-build/

```
FROM ros:indigo as base
...

FROM base as buggy
...
```

Comment (afsafzal) [reply test/hello_world/Dockerfile]: You mean having one Dockerfile instead of two?

**Pull request merged**